Cast subject page to a Title if it isn't one

SomeMWDev opened the pull request:
Downstream task: https://issue-tracker.miraheze.org/T14342

ContentParseParams::getPage returns a PageReference.
This fixes:
```
Maps\MapsFactory::newSemanticGeoJsonStore(): Argument #2 ($subjectPage) must be of type MediaWiki\Title\Title, MediaWiki\Page\PageIdentityValue given, called in /srv/mediawiki/1.44/extensions/Maps/src/GeoJsonPages/GeoJsonContentHandler.php on line 51

from /srv/mediawiki/1.44/extensions/Maps/src/MapsFactory.php(238)
#0 /srv/mediawiki/1.44/extensions/Maps/src/GeoJsonPages/GeoJsonContentHandler.php(51): Maps\MapsFactory->newSemanticGeoJsonStore(MediaWiki\Parser\ParserOutput, MediaWiki\Page\PageIdentityValue)
#1 /srv/mediawiki/1.44/includes/content/ContentHandler.php(1692): Maps\GeoJsonPages\GeoJsonContentHandler->fillParserOutput(Maps\GeoJsonPages\GeoJsonContent, MediaWiki\Content\Renderer\ContentParseParams, MediaWiki\Parser\ParserOutput)
#2 /srv/mediawiki/1.44/includes/content/Renderer/ContentRenderer.php(75): MediaWiki\Content\ContentHandler->getParserOutput(Maps\GeoJsonPages\GeoJsonContent, MediaWiki\Content\Renderer\ContentParseParams)
#3 /srv/mediawiki/1.44/includes/Revision/RenderedRevision.php(260): MediaWiki\Content\Renderer\ContentRenderer->getParserOutput(Maps\GeoJsonPages\GeoJsonContent, MediaWiki\Page\PageIdentityValue, MediaWiki\Revision\RevisionStoreRecord, MediaWiki\Parser\ParserOptions, array)
#4 /srv/mediawiki/1.44/includes/Revision/RenderedRevision.php(233): MediaWiki\Revision\RenderedRevision->getSlotParserOutputUncached(Maps\GeoJsonPages\GeoJsonContent, array)
#5 /srv/mediawiki/1.44/includes/Revision/RevisionRenderer.php(236): MediaWiki\Revision\RenderedRevision->getSlotParserOutput(string, array)
#6 /srv/mediawiki/1.44/includes/Revision/RevisionRenderer.php(169): MediaWiki\Revision\RevisionRenderer->combineSlotOutput(MediaWiki\Revision\RenderedRevision, MediaWiki\Parser\ParserOptions, array)
#7 /srv/mediawiki/1.44/includes/Revision/RenderedRevision.php(196): MediaWiki\Revision\RevisionRenderer->MediaWiki\Revision\{closure}(MediaWiki\Revision\RenderedRevision, array)
#8 /srv/mediawiki/1.44/includes/poolcounter/PoolWorkArticleView.php(110): MediaWiki\Revision\RenderedRevision->getRevisionParserOutput()
#9 /srv/mediawiki/1.44/includes/poolcounter/PoolWorkArticleView.php(81): MediaWiki\PoolCounter\PoolWorkArticleView->renderRevision()
#10 /srv/mediawiki/1.44/includes/poolcounter/PoolCounterWork.php(173): MediaWiki\PoolCounter\PoolWorkArticleView->doWork()
#11 /srv/mediawiki/1.44/includes/page/ParserOutputAccess.php(364): MediaWiki\PoolCounter\PoolCounterWork->execute()
#12 /srv/mediawiki/1.44/includes/page/Article.php(834): MediaWiki\Page\ParserOutputAccess->getParserOutput(MediaWiki\Page\WikiPage, MediaWiki\Parser\ParserOptions, MediaWiki\Revision\RevisionStoreRecord, int)
#13 /srv/mediawiki/1.44/includes/page/Article.php(551): MediaWiki\Page\Article->generateContentOutput(MediaWiki\User\User, MediaWiki\Parser\ParserOptions, int, MediaWiki\Output\OutputPage, array)
#14 /srv/mediawiki/1.44/includes/actions/ViewAction.php(80): MediaWiki\Page\Article->view()
#15 /srv/mediawiki/1.44/includes/actions/ActionEntryPoint.php(728): MediaWiki\Actions\ViewAction->show()
#16 /srv/mediawiki/1.44/includes/actions/ActionEntryPoint.php(505): MediaWiki\Actions\ActionEntryPoint->performAction(MediaWiki\Page\Article, MediaWiki\Title\Title)
#17 /srv/mediawiki/1.44/includes/actions/ActionEntryPoint.php(143): MediaWiki\Actions\ActionEntryPoint->performRequest()
#18 /srv/mediawiki/1.44/includes/MediaWikiEntryPoint.php(202): MediaWiki\Actions\ActionEntryPoint->execute()
#19 /srv/mediawiki/config/initialise/entrypoints/index.php(71): MediaWiki\MediaWikiEntryPoint->run()
#20 {main}
```

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **Bug Fixes**
  * More reliable GeoJSON saving by normalizing page identity before storage, reducing failures and errors when creating or updating map data.
* **Chores**
  * Updated extension compatibility: minimum required platform version raised from 1.40.0 to 1.43.0.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

SomeMWDev commented:
Raised the MW requirement to 1.41 since `Title::newFromPageReference` was added in that version and 1.40 is EOL anyway (1.41 is too, for that matter, so in theory it should be raised to 1.43)

JeroenDeDauw commented:
Better to bump to MW 43 right away then, since every time we increase the needed MW version, we need  a major release

https://github.com/ProfessionalWiki/Maps/blob/master/INSTALL.md

SomeMWDev commented:
> Better to bump to MW 43 right away then, since every time we increase the needed MW version, we need a major release
> 
> https://github.com/ProfessionalWiki/Maps/blob/master/INSTALL.md

Done, also removed the redundant PHP requirement and updated the CI config.
